### ISSUE TITLE
URISyntaxException in the OIDC flow when using the form_post respond mode

### ISSUE THREAD

**Issue by somindatommy** [wso2/product-is#9152]
**Describe the issue:**
When using the code grant with the form_post respond mode, the following error is experienced.

```
TID: [-1234] [] [https-jsse-nio-9443-exec-50] [264] [2020-08-04 15:19:58,693] [cb8103b6-6d39-40a0-b655-53b4940078ab] ERROR {org.wso2.carbon.identity.oauth.endpoint.authz.OAuth2AuthzEndpoint} - Error while getting authorization code from redirectURL java.net.URISyntaxException: Illegal character in scheme name at index 0: {"code":"353dbdf8-b19e-3a66-b0cd-e4244ea02db6","state":"OpenIdConnect.AuthenticationProperties=m3cNphJG8tSJU376zRKuw_bKj9feK4E5ndWrIagct0jCF_dXXL53wGaXn4HcUrYO9wXebXaBQsuQKJponmq8Rus4VxPMyplljo08MAQrPYXqZecmQ30GDYvffJlmYRD9-Cj6B_60juEg6ENBRsGhZxQAO3cCZUOeqhn7LQqcb-x9ix5xnlY2XF7lIhSPzSXFQ7ZfWiHrGvg2cfkoOJ-mV86sCmr8cIQhlZdOh7ynnfgfJKfRJLSrowPGPzLyh2Xx"}
        at java.net.URI$Parser.fail(URI.java:2848)
        at java.net.URI$Parser.checkChars(URI.java:3021)
        at java.net.URI$Parser.checkChar(URI.java:3031)
        at java.net.URI$Parser.parse(URI.java:3047)
        at java.net.URI.<init>(URI.java:588)
        at org.apache.http.client.utils.URIBuilder.<init>(URIBuilder.java:81)
        at org.wso2.carbon.identity.oauth.endpoint.authz.OAuth2AuthzEndpoint.getAuthCodeFromRedirectURL(OAuth2AuthzEndpoint.java:2834)
        at org.wso2.carbon.identity.oauth.endpoint.authz.OAuth2AuthzEndpoint.storeSidClaim(OAuth2AuthzEndpoint.java:2726)
        at org.wso2.carbon.identity.oauth.endpoint.authz.OAuth2AuthzEndpoint.manageOIDCSessionState(OAuth2AuthzEndpoint.java:2539)
        at org.wso2.carbon.identity.oauth.endpoint.authz.OAuth2AuthzEndpoint.handleFormPostMode(OAuth2AuthzEndpoint.java:792)
        at org.wso2.carbon.identity.oauth.endpoint.authz.OAuth2AuthzEndpoint.handleSuccessfulAuthentication(OAuth2AuthzEndpoint.java:716)
        at org.wso2.carbon.identity.oauth.endpoint.authz.OAuth2AuthzEndpoint.handleAuthenticationResponse(OAuth2AuthzEndpoint.java:677)
        at org.wso2.carbon.identity.oauth.endpoint.authz.OAuth2AuthzEndpoint.authorize(OAuth2AuthzEndpoint.java:246)
        at sun.reflect.GeneratedMethodAccessor223.invoke(Unknown Source)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.apache.cxf.service.invoker.AbstractInvoker.performInvocation(AbstractInvoker.java:179)
        at org.apache.cxf.service.invoker.AbstractInvoker.invoke(AbstractInvoker.java:96)
        at org.apache.cxf.jaxrs.JAXRSInvoker.invoke(JAXRSInvoker.java:193)
```

This issue is partly fixed in https://github.com/wso2/product-is/issues/5995. We have only fixed for 
```
if(redirectURL.contains(ID_TOKEN)
```
https://github.com/wso2-support/identity-inbound-auth-oauth/blob/b0ee8c721631468fbb3e85003875c9facd6070f0/components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/authz/OAuth2AuthzEndpoint.java#L2723

**How to reproduce:**
 -  Create an OIDC service provider.
 -  Use the following link to reproduce. (replace client_id and redirect_url)

```
https://localhost:9443/oauth2/authorize?response_type=code&client_id=5vJaavcK5QOA1Bq5vJZAwc43kk0a&redirect_uri=https://example-app.com/redirect&scope=openid&response_mode=form_post
```

**Reported version**
- IS 5.10.0

**Comment by janakamarasena** [comment]
Possibly fixed from https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1402

**Comment by GANGANI** [comment]
This issue was already fixed by https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1402. Hence this is not reproducible in public or IS-5.11.0. This was also fixed in IS-5.8.0 with https://github.com/wso2-support/identity-inbound-auth-oauth/pull/818.

**Comment by NilukaSripalim** [comment]
Tested in stage
